### DISCUSSION TITLE
using the TRDA-ae design system with React. Is possible ?

### DISCUSSION THREAD

**Discussion by x0341** [TDRA-ae/aegov-dls#32] [Q&A]
Is possible to use the TRDA-ae design system with React. 
can you provide a sample ?

thanks.

**Comment by moiz-conceptlz** [comment]
ReactJS is a library, but the underlying output of any React component is HTML with attributes and predefined CSS.

Although there is no React package at the moment published by the UAE Design System, let me help you in terms of how you may use ReactJS with the design system.

For the purpose of the example, I'll be using the Button component.

To place a button on the website, the code basic example provided on https://designsystem.gov.ae/docs/components/button is:
```html
<button class="aegov-btn" type="button">Base button</button>
```

Now let's see how you can convert this into a React component. You would essentially add a component, like this:
```js
import React from 'react';

const Button = ({ children, className }) => {
return (
    <button class="{className}" type="button">
      {children}
    </button>
  );
};

export default Button;
```

If you really want to ensure the button is enforced with the design system attribute, you can also do something like this, in which case, you do not need to pass the `aegov-btn` class to the component, but rather have it embedded in the render by default and only pass additional classes when required.
```js
import React from 'react';

const Button = ({ children, className }) => {
return (
    <button class="{`aegov-btn ${className}`}" type="button">
      {children}
    </button>
  );
};

export default Button;
```

Now, how about handling the click events:
```js
import React from 'react';

const Button = ({ children, className, onClick }) => {
return (
    <button class="{`aegov-btn ${className}`}" type="button" onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
```

To use the above in your app, you would do something like this:
```js
import React from 'react';
import Button from './Button';

const App = () => {
  const handleClick = () => {
    console.log('Button was clicked!');
  };

  return (
    <div>
      <Button onClick={handleClick} classes='btn-outline'>
        Click Me!
      </Button>
    </div>
  );
};

export default App;
```

In the code above, we imported the button component and used it in the App component. We passed an onClick prop with a function handleClick that will be called when the button is clicked.

As you can see, the design system does not stop you from using any JS framework. What matters is the rendered output HTML, and the use of the pre-defined CSS classes. 

Hope this was helpful.

**Comment by moiz-conceptlz** [comment]
@x0341 I hope you have voted on https://github.com/orgs/TDRA-ae/discussions/40
